shell-setup/src/environment.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * A collection of functions that interact with the environment, to allow
+ * for potentially mocking in tests in the future.
+ */
+import { which } from "@david/which";
+import { homedir as getHomeDir } from "node:os";
+
+async function tryStat(path: string): Promise<Deno.FileInfo | undefined> {
+  try {
+    return await Deno.stat(path);
+  } catch (error) {
+    if (
+      error instanceof Deno.errors.NotFound ||
+      (error instanceof Deno.errors.PermissionDenied &&
+        (await Deno.permissions.query({ name: "read", path })).state ==
+          "granted")
+    ) {
+      return;
+    }
```

Comment:
Running with `-A` would make this code not required?

Reply:
Yeah that's true. I'd prefer to leave it in though, so that this script works regardless of the permission flags